features/foutafhandeling.feature

```diff
@@ -0,0 +1,542 @@
+# language: nl
+
+Functionaliteit: Als Consumer van de BAG API wil ik consistente en gebruiksvriendelijke foutresponses conform problem+json
+    Zodat ik in mijn consumer applicatie de fouten eenduidig kan afhandelen.
+
+Rule: een verplichte parameter moet zijn opgegeven met een waarde
+
+    Scenario: Verplichte parameter is niet opgegeven
+        Als '/adressen/zoek' wordt aangeroepen
+        Dan bevat de response ten minste de volgende velden
+        | naam   | waarde                                                                            |
+        | title  | Ten minste één parameter moet worden opgegeven.                                   |
+        | status | 400                                                                               |
+        | code   | paramsRequired                                                                    |
+        En bevat de response geen veld "invalidParams"
+        En is de http status code van het antwoord 400
+        En bevat de response de header "Content-Type" met waarde "application/problem+json"
+
+Rule: alleen gespecificeerde queryparameters mogen worden gebruikt
+
+    Scenario: Niet gespecificeerde parameter
+        Als '/adressen?bestaatniet=abcd' wordt aangeroepen
+        Dan bevat de response ten minste de volgende velden
+        | naam   | waarde                                        |
+        | title  | Een of meerdere parameters zijn niet correct. |
+        | status | 400                                           |
+        | code   | paramsValidation                              |
+        En bevat de response de volgende invalidParams
+        | name        | code         | reason                      |
+        | bestaatniet | unknownParam | Parameter is niet verwacht. |
+
+Rule: de parameterwaarde moet aan de parameterspecificaties voldoen
+
+    Abstract Scenario: waarde van een queryparameter is niet correct
```

Comment:
'Niet correct' is niet specifiek genoeg. Is een waarde niet correct omdat het type van de waarde niet overeenkomt met het gespecificeerde type (rule op regel 89) of omdat er een waarde is opgegeven die niet binnen de grenzen zoals gespecificeerd in de API specificatie valt (rule op regel 105)?
Volgens mij zijn regel; 47, 48, 49 type fouten  en regel: 50, 51, 52, 53, 54, 55 validatie fouten.

Reply:
maakt dat hier uit? Het gaat erom dat verschillende (soorten) fouten aan parameterwaarden worden gemeld. Dit is wat in de common foutafhandeling staat bij regel 49 e.v.

Ik maak ervan "waarde van een queryparameter is niet correct conform parameterspecificaties"